### DISCUSSION TITLE
What is the poetry equivalent to `pip install -e .[dev]`

### DISCUSSION THREAD

**Discussion by mikel-brostrom** [python-poetry/poetry#8263] [Q&A] [answered]
Let's say I have this:

```
extras_require={
        'dev': [
            'pytest',
            'pytest-cov',
            'coverage',
            'pre-commit'
        ],
    }
```

in my `setup.py`. In order to install these dependencies I would run: `pip install -e .[dev]`. What is the equivalent of all this in poetry?

**Comment by radoering** [comment, marked as answer]
If it's your current project you want to install, it's `poetry install -E dev`

If it's another dependency you want to add to your project, it's `poetry add -e -E dev <path>`

**Reply by mikel-brostrom** [reply]
Thanks!
